packages/crypto/src/bls12_318/pairing.rs

```diff
@@ -0,0 +1,171 @@
+use crate::CryptoError;
+
+use super::points::{g1_from_variable, g2_from_variable};
+use bls12_381::G2Prepared;
+use pairing::group::Group;
+use rayon::iter::{ParallelBridge, ParallelIterator};
+
+pub fn bls12_381_aggregate_pairing_equality(
+    ps: &[u8],
+    qs: &[u8],
+    r: &[u8],
+    s: &[u8],
+) -> Result<bool, CryptoError> {
+    let pq_pairs: Vec<_> = ps
+        .chunks_exact(48)
+        .zip(qs.chunks_exact(96))
```

Comment:
I think for those decodings we should have explicit error messages along the lines of:
- "first argument must not be empty"
- "second argument must not be empty"
- "first argument is not a multiple of 48 bytes: {} bytes"
- "second argument is not a multiple of 96 bytes: {} bytes"
- "Number of points in first argument ({}) is different from number of points in second argument ({})"

Please improve those texts, but I guess you get the idea. We have a lot of context for what's going wrong when the API is misused.

Reply:
Definitely. The current iteration is more of a "get it to work" thing, improving the error messages would have been part of the polishing up process of the PR.  
Will do!

Reply:
The errors returned now look something like this: https://github.com/CosmWasm/cosmwasm/blob/868c49bfa12d37f3c63761b30b836f577d213772/packages/core/src/errors/verification_error.rs#L9-L18

Ugh, I just realized I missed the "list empty" case. Adding that, too..